pipeline/lib/metric.ml

```diff
@@ -0,0 +1,313 @@
+      job _worker_job_id () =
+    let open Lwt.Infix in
+    Current.Job.start job ~level:Current.Level.Above_average >>= fun () ->
+    Logs.info (fun log -> log "Looking for significantly changed metrics...");
+    match
+      Storage.get_main_branch_metrics ~conninfo ~repository ~worker
+        ~docker_image
+    with
+    | Some result_main, Some result -> (
+        let compare_metrics, main_commit =
+          parse_benchmark_data ~data:result_main
+        in
+        let metrics, _ = parse_benchmark_data ~data:result in
+        let changes = find_changed_metrics ~metrics ~compare_metrics in
+        match github_app with
+        | Some app when not (BenchmarksData.is_empty changes) ->
+            let text =
+              format_changes_message ~changes ~repository ~worker ~docker_image
+                ~main_commit
+            in
+            let open Current.Syntax in
+            let _ =
+              let+ installations = Github.App.installations app in
+              let apis = List.map Github.Installation.api installations in
+              let api = match apis with [] -> None | api :: _ -> Some api in
+              match api with
+              | Some api ->
+                  post_github_comment ~api ~repository ~pull_number ~text
+              | None -> ()
+            in
+            Lwt.return (Ok text)
+        | _ -> Lwt.return (Ok "No Github app configured or no changes found"))
+    | _ -> Lwt.return (Ok "No results found for comparison")
+end
+
+module NGC = Current_cache.Output (NotifyGithub)
+
+let notify ~conninfo ~repository ~github_app ~worker ~docker_image ~pull_number
+    job_id =
+  let open Current.Syntax in
+  Current.component "notify-github"
+  |> let> job_id in
+     match job_id with
+     | None -> Current_incr.const (Error (`Active `Ready), None)
+     | Some job_id ->
+         NGC.set
+           {
+             conninfo;
+             repository;
+             github_app;
+             worker;
+             docker_image;
+             pull_number;
+           }
+           job_id ()
+
+let notify_metric_changes ~conninfo ~repository ~worker ~docker_image
+    ~github_app ~config =
+  let repo_id = Repository.info repository in
+  let commit_hash = Repository.commit_hash repository in
+  let job_id = Fmt.str "%s:%s" repo_id commit_hash in
+  let repo_configs = Config.find config repo_id in
+  let should_notify =
+    List.find_map
+      (fun (repo_config : Config.repo) ->
+        match repo_config.notify_github with
+        | true
+          when worker = repo_config.worker && docker_image = repo_config.image
+          ->
+            Some repo_config
+        | _ -> None)
+      repo_configs
+  in
+  let pull_number = Repository.pull_number repository in
+  match (pull_number, should_notify) with
+  | Some pull_number, Some _ ->
+      let job_id = Current.return (Some job_id) in
+      let _ =
+        notify ~conninfo ~repository ~github_app ~worker ~docker_image
```

Comment:
Given my understanding of ocurrent, I think the `let _ =` works, but is a bit sneaky! (a `Current.t` node is attached to the pipeline, yet the type of `notify_metric_changes` doesn't reflect it :/ ). Could we not ignore the outcome? (so that we would be able to report any errors to slack in this stage?)

Reply:
(... but if this is changed, there's some extra-difficulties in order to not (?) send an enormous message when the execution crashed and produced no json!)

Reply:
I'm not sure what the best way out here is. I pushed a commit to use an `ignore` instead of `let _`, but not sure if that's what we want here.

Reply:
Ha but it's very much the same as before with `ignore` :P 
If `job_id` was defined as dependent on the worker output, then it wouldn't be available until the worker terminates (something like `let job_id = Current.map (fun _ -> job_id) worker_output`) ... which would allow to `and+ () = Metric.notify...` rather than `ignore` :)

Reply:
Ah, yes. That makes sense! Let me try that.

Reply:
I think my current change should be close to what you suggest. Could you take a look again? :)